lib/puppet/pops/loader/loader_paths.rb

```diff
@@ -74,6 +74,16 @@ def root_path
       @loader.path
     end
 
+    # Answers `true` if the loader used by this instance is the system_loader. The system_loader
+    # will have a path relative to the parent of 'puppet' instead of the parent of 'lib/puppet'
+    # since the 'lib' directory of puppet is renamed during install.
+    # This is significant for loaders that load ruby code.
+    #
+    # @return [Boolean] a boolean answering if the loader is the system loader.
+    def system_loader?
+      @loader.module_name.nil?
```

Comment:
This does not seem like a very robust way of determining if a loader is a system loader.